lib/prepare-write.js

```diff
@@ -44,7 +44,9 @@ function prepareWrite(outFolder, file, opt, callback) {
   file.stat.mode = options.mode;
   file.flag = options.flag;
   file.cwd = cwd;
-  file.base = basePath;
+  // Ensure the base always ends with a separator
+  // TODO: add a test for this
+  file.base = path.normalize(basePath + path.sep);
```

Comment:
since `path.normalize()` will fix `'foo//bar'` to `foo/bar` (or backslashes if windows), to be safe I would consider doing `basePath + '/' + path.sep`. Or just `path.join()`, but if you're doing normalize join isn't really needed.